### DISCUSSION TITLE
Running headless

### DISCUSSION THREAD

**Discussion by ajvdw** [GrandOrgue/grandorgue#685] [General]
I am looking to a proper solution to run GrandOrgue without a screen on a Raspberry PI 4. Loosing all the UI stuff would free up some resources as well. What's the best approach? I also want to send some commands to the core to store some settings before shutting down properly.

**Comment by oleg68** [comment]
1. You may connect a monitor, run GO, make settings, store them and close GO
2. Next time you may run GO without a screen.

But I doubt that GO can ever run without a screen.

**Comment by ajvdw** [comment]
I found out how to run headless on a raspberry pi:
- Install VNC via raspi-config.
- You can start GO via terminal if you entered the command "export DISPLAY=:0.0" first.
- This brings another advantage, you can define the display size as you like; this solves most visualisation issues as well.
- I'm running on the 64 bit edition with a hifiberry with minor issues. 
- The major advantage is that I now use an iPad accessing the Pi through VNC (touch!).

**Comment by oleg68** [comment]
@ajvdw 

How did you start the VNC session?

The main problem I see is when a user connects with VNC its session is considered as remote, and systemd does not allow the user to access to sound cards, so GO cann't work.

There are several workarounds (change the default device permisions, run GO by root, run jackd as root and so on), but all of they are not nice

**Comment by ajvdw** [comment]
Hi,

I did nothing special.
Installed vnc via rasps-config.
Make GO to start automatically.

And it runs directly after boot without a screen connected.

I see some ALSA messages when GO is running, but they also appear when running non-headless. I assumed that they were cause by a problematic ALSA implementation on the Pi and the fact that PortAudio is more than 10 years unchanged and some API calls are deprecated (as indicated during build).

AJ

**Comment by oleg68** [comment]
> Make GO to start automatically.

By what user?. Are you using the auto-logon feature?

>  see some ALSA messages when GO is running, but they also appear when running non-headless. I

Please use the last release from GitHub. It is shipped with a last RtAudio release (4.0.0). You can try to switch the output device to the `Rt` port instead of `Pa`.

>  the fact that PortAudio is more than 10 years unchanged and some API calls are deprecated (as indicated during build).

Please create a separate issue about PA implementation with the exact error message.

**Comment by ajvdw** [comment]
No Auto-logon, user PI without elevated right (out of the box). The shellscript runs fro
Deskkop Autostart. I did the following:

```
Mkdir /home/pi/.config/autostart/
Create file grandorgue.desktop
[Desktop Entry]
Name=GrandOrgue
Exec=/home/pi/GO/startGO
Type=Application

```
The startGO script look like:

```
export DISPLAY=:0.0
GrandOrgue &

```

I used the last raspberry PI build from GitHub (grandorgue_0.3.1.2341-12.os_arm64.deb)


AJ

**Comment by oleg68** [comment]
> No Auto-logon, user PI without elevated right (out of the box)

What does ``loginctl`` command return?

> The startGO script look like:

How do you launch this script?

**Comment by oleg68** [comment]
> I used the last raspberry PI build from GitHub (grandorgue_0.3.1.2341-12.os_arm64.deb)

1. Do you have an issue #71 with this release?
2. Does the ALSA error message still appear with Rt port?

**Comment by ajvdw** [comment]
SESSION  UID USER SEAT  TTY  
      1 1000 pi   seat0 tty1 
      3 1000 pi   seat0      
      4 1000 pi         pts/0


> Op 20 aug. 2021, om 11:27 heeft Oleg Samarin ***@***.***> het volgende geschreven:
> 
> loginctl

**Comment by oleg68** [comment]
Yes. Raspbery has the auto-logon feature out of box, when other linux distributions require a special setup for it. So the user `pi` always has access to the audio.
